File: java/src/acme/application/command/CreateUserCommand.java
package acme.application.command;

import sharedkernel.application.command.Command;

public class CreateUserCommand implements Command {
    private String email;
    private String password;
    private String firstName;
    private String lastName;

    public CreateUserCommand(String email, String password, String firstName, String lastName) {
        this.email = email;
        this.password = password;
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String email() {
        return email;
    }

    public String password() {
        return password;
    }

    public String firstName() {
        return firstName;
    }

    public String lastName() {
        return lastName;
    }
}

File: kotlin/src/acme/application/command/CreateUserCommand.kt
package acme.application.command

import sharedkernel.application.command.Command

class CreateUserCommand(
    private val email: String,
    private val password: String,
    private val firstName: String,
    private val lastName: String
) : Command {

    fun email(): String {
        return email
    }

    fun password(): String {
        return password
    }

    fun firstName(): String {
        return firstName
    }

    fun lastName(): String {
        return lastName
    }
}
